sys/mips/mips/vm_machdep.c

```diff
@@ -578,28 +583,49 @@ swi_vm(void *dummy)
 }
 
 int
-cpu_set_user_tls(struct thread *td, void *tls_base)
+cpu_set_user_tls(struct thread *td, void * __capability tls_base)
 {
 
-#if defined(__mips_n64) && defined(COMPAT_FREEBSD32)
-	if (td->td_proc && SV_PROC_FLAG(td->td_proc, SV_ILP32))
-		td->td_md.md_tls_tcb_offset = TLS_TP_OFFSET + TLS_TCB_SIZE32;
+#ifdef COMPAT_FREEBSD32
+	if (SV_PROC_FLAG(td->td_proc, SV_ILP32))
+		td->td_md.md_tls_tcb_offset = TLS_TP_OFFSET32 + TLS_TCB_SIZE32;
 	else
 #endif
-#if defined (COMPAT_CHERIABI)
-	/*
-	 * XXX-AR: should cheriabi_set_user_tls just delegate to this
-	 * function?
-	 */
-	if (td->td_proc && SV_PROC_FLAG(td->td_proc, SV_CHERI))
-		panic("cpu_set_user_tls(%p) should not be called from CHERIABI\n", td);
+#ifdef COMPAT_FREEBSD64
+	if (!SV_PROC_FLAG(td->td_proc, SV_CHERI))
```

Comment:
Perhaps better as `SV_PROC_FLAG(td->td_proc, SV_LP64 | SV_CHERI) == SV_LP64` like is done elsewhere in the kernel for freebsd64?

Reply:
Done